### DISCUSSION TITLE
how to get a thumbnail

### DISCUSSION THREAD

**Discussion by qiangwai** [donmccurdy/glTF-Transform#547] [Q&A] [answered]
I want generate a thumbnail in node by this SDK, have any good idea?

**Comment by donmccurdy** [comment, marked as answer]
Hi! The gltf-transform library is for reading/editing/writing glTF files and doesn't really deal with rendering<sup>1</sup>. You'll need a renderer for that. Because node.js is a server-side platform it isn't particularly rich in rendering libraries, but you can spawn headless Blender instances or things like that if needed. 

I'd start with this project: https://github.com/Shopify/screenshot-glb

<sup>1</sup> There is also the [`@gltf-transform/view` module](https://github.com/donmccurdy/glTF-Transform-View), which could connect with three.js to render glTF files, but (1) running three.js in Node.js isn't trivial, and (2) three.js can already load glTF files so you don't really need gltf-transform for that workflow.
